src/test/java/core/basesyntax/service/impl/FileWriterImplTest.java

```diff
@@ -0,0 +1,54 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
+import core.basesyntax.service.operations.report.FileWriter;
+import core.basesyntax.service.operations.report.impl.FileWriterImpl;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class FileWriterImplTest {
+    private static final String TEST_FILE_PATH = "src/test/resources/test_report.txt";
+    private static final String TEST_CONTENT = "This is a test report content.";
+    private FileWriter fileWriter;
+
+    @BeforeEach
+    void setUp() {
+        fileWriter = new FileWriterImpl();
+    }
+
+    @AfterEach
+    void tearDown() throws IOException {
+        Files.deleteIfExists(Path.of(TEST_FILE_PATH));
+    }
+
+    @Test
+    void writeDataToFile_ValidFilePath_ok() throws IOException {
+        fileWriter.writeDataToFile(TEST_CONTENT, TEST_FILE_PATH);
+        File file = new File(TEST_FILE_PATH);
+        assertTrue(file.exists());
+        List<String> actualLines = Files.readAllLines(Path.of(TEST_FILE_PATH));
+        assertEquals(1, actualLines.size());
+        assertEquals(TEST_CONTENT, actualLines.get(0));
+    }
+
+    @Test
+    void writeDataToFile_NullFilePath_notOk() {
+        assertThrows(NullPointerException.class, () ->
+                fileWriter.writeDataToFile(TEST_CONTENT, null));
+    }
+
+    @Test
+    void writeDataToFile_NullContent_notOk() {
+        assertThrows(RuntimeException.class, () ->
```

Comment:
Let's replace all raw Runtime exceptions with a custom ones, in both impl and test methods, because NPE for example is also a Runtime, and so we expect any exception here really, which  is not good
And also assert the exception message